### PR TITLE
Morna, Travis!

### LINKED ISSUE #424: Change from Travis to GitHub actions?
Travis CI is cutting down on their free offering:

https://news.ycombinator.com/item?id=25338983

We better switch to something different before they cut service entirely.

### PR DESCRIPTION
**Pull request by mbakke**
These commits gets rid of Travis entirely in favor of a new GitHub action.

See #444 for context.

Fixes #424.

I'm not sure how to add the test as a branch restriction, but will look into it after the merge.  I've removed the Travis requirement meanwhile.

Edit: it looks like the new action checks already restrict merging!

### CODE REVIEW

**Comment by oyvindkolbu** [comment]
Good stuff! Needs to check why the upload to coveralls fails.

**Review by oyvindkolbu — APPROVED** [review]
With or without fixing coveralls, this is a huge improvement! :shipit:

**Comment by mbakke** [comment]
I disabled the branch protection option that dismisses reviews when a new commit is pushed, because I abuse `git commit --amend` a lot...

Anyway, here is a new version that adds a separate step for Coveralls, uploading only the Python 3.9 results for simplicity.  I haven't tested it yet, because it only runs when new commits are added on `master`, as the built-in `GITHUB_TOKEN` magic apparently does not work with forks.

I'm not entirely convinced using the GitHub token will work, as some documentation suggested it only worked for "repository owners", and I'm not sure what that means in our case.  But we can always add the proper API token (accessible from [coveralls.io](https://coveralls.io/github/unioslo/mreg)) as a secret if using the GitHub token fails.

That is to say, I think this is ready to merge, and I can fix up Coveralls on 'master' by meddling with the branch protection settings...

@oyvindkolbu thoughts on uploading Python 3.9 results only?  Not sure how to achieve the multi-job magic added by Travis to the coveralls report.

**Comment by mbakke** [comment]
I was able to test it on my personal repo by using the Coveralls token as a secret.

However, downloading the `.coverage` file does not work:

https://github.com/mbakke/mreg/runs/2915978748?check_suite_focus=true

Got to go now, but will fix before merging.

**Comment by mbakke** [comment]
Coveralls is working now :tada: 

https://github.com/mbakke/mreg/runs/2917119432?check_suite_focus=true

(the `ls` step is removed in this PR)

**Comment by oyvindkolbu** [comment]
> @oyvindkolbu thoughts on uploading Python 3.9 results only? Not sure how to achieve the multi-job magic added by Travis to the coveralls report.

The coverage should be same with all python versions, so probably not an issue.

**Review by oyvindkolbu — APPROVED** [review]

**Review by oyvindhagberg — APPROVED** [review]
Good work! Looks good to me.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
+on: [push, pull_request]
+
+name: CI
+jobs:
+  test:
+    name: Test
+    runs-on: ${{ matrix.os }}
+    services:
+      postgres:
+        image: postgres:11
```
[the remark this answers is not in the document]
Comment (mbakke) [reply .github/workflows/test.yml]: I switched to `latest` which is currently an alias for 13.

**Comment by mbakke** [comment]
I figured out how to do multi-job coverage reports and pushed a version that, according to all available documentation, should work -- even for PRs.  However there are apparent issues with the Coveralls GitHub integration, and also the third-party [`python-coveralls`](https://github.com/TheKevJames/coveralls-python) package, so expect a few follow-up PRs to resolve the situation.

**Pull request merged**